x/stake/types/validator.go

```diff
@@ -404,15 +409,11 @@ func (v Validator) AddTokensFromDel(pool Pool, amount sdk.Int) (Validator, Pool,
 
 // RemoveDelShares removes delegator shares from a validator.
 func (v Validator) RemoveDelShares(pool Pool, delShares sdk.Dec) (Validator, Pool, sdk.Dec) {
-	issuedTokens := v.DelegatorShareExRate().Mul(delShares)
-	v.Tokens = v.Tokens.Sub(issuedTokens)
+	delTokens := v.DelegatorShareExRate().Mul(delShares)
+	delTokens = sdk.MinDec(delTokens, v.Tokens)
```

Comment:
Why is this necessary? When are there insufficient tokens? Would that mean some other delegator got more than they ought to?

Reply:
All this logic is updated in https://github.com/cosmos/cosmos-sdk/pull/2958 p.s. as mentioned in the previous comment